### PR TITLE
Update Gradle wrapper to 3.5

### PR DESCRIPTION
**Pull request by friederbluemle**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -202,8 +202,3 @@ buildscript {
         classpath 'org.eclipse.jgit:org.eclipse.jgit:2.0.0.201206130900-r'
     }
 }
-
-task wrapper(type: Wrapper) {
```
Comment (JesusFreke) [inline build.gradle]: Hmm. Why are you deleting this?

The usual way I upgrade is to update the version here and then do a "./gradlew wrapper" (sometimes twice? *shrug*). How did you perform the upgrade?

Comment (friederbluemle) [inline build.gradle]: Newer version of Gradle have the `wrapper` task *already included*. Upgrading to a new Gradle wrapper version is as easy as invoking this command:

    ./gradlew wrapper --gradle-version 3.5 --distribution-type all

Note: The distribution type defaults to `bin`, and for Android projects the default and recommended distribution is `all`, as it adds additional support (sources) to the IDE while working on Gradle scripts. The version/distribution only needs to be downloaded *once* for all projects and is cached system-wide (usually in `~/.gradle`).

Comment (JesusFreke) [inline build.gradle]: Gotcha, thanks! Honestly, I perform the gradle upgrade so rarely that I'm never going to be able to remember the exact parameter names, etc. Let's just keep the task there, for future convenience :)

Comment (friederbluemle) [inline build.gradle]: @JesusFreke Okay, I added a comment with the command line syntax for the Gradle Wrapper upgrade in `build.gradle`. Would you be OK with that?

I don't think it's a good idea to override the included `wrapper` task with a custom one with the same name:
- It removes the documentation and any (potentially) added functionality of the included one.
- It requires to open, edit, *and (unnecessarily) commit* `build.gradle` for each wrapper upgrade.
- The Gradle wrapper version has to be kept in sync in both `build.gradle` **and** `gradle/wrapper/gradle-wrapper.properties`

Comment (JesusFreke) [inline build.gradle]: See [gradle's recommendation](https://docs.gradle.org/current/userguide/gradle_wrapper.html)

>     The Wrapper can be further customized by adding and configuring a Wrapper task in your build script, and then executing it.


1. No it doesn't :)
2. Upgrading gradle requires a commit anyway
3. Keeping these in sync is trivial. When performing a gradle upgrade, you simply change the version in build.prop and run the task.

Comment (friederbluemle) [inline build.gradle]: Of course it requires a commit - It doesn't require a modification to `build.gradle` is what I was saying. :)

Do you want me to add the custom `wrapper` task back in?

Comment (JesusFreke) [inline build.gradle]: Yeah, let's keep it :)

Comment (friederbluemle) [inline build.gradle]: Sure thing. I added it back! :)

**Pull request merged**